src/Input.cpp

```diff
@@ -150,28 +150,40 @@ struct Gosu::Input::Impl
                     break;
                 }
                 int gamepad_slot = -1;
-                int device_instance_id = -1;
-                // Prefer the SDL_GameController API...
-                if (SDL_IsGameController(e->jdevice.which)) {
-                    if (SDL_GameController *game_controller = SDL_GameControllerOpen(e->jdevice.which)) {
-                        open_game_controllers.emplace_back(
-                            shared_ptr<SDL_GameController>(game_controller, SDL_GameControllerClose)
-                        );
+                int joystick_instance_id = -1;
+
+                // Loop through attached gamepads as e->jdevice.which cannot be trusted (always 0)
```

Comment:
When I start a Gosu game with two gamepads already attached on macOS, `e->jdevice.which` is 0 for the first gamepad and 1 for the second. I wonder if this is actually an SDL bug on Linux?